frontend/src/component/common/PrettifyLargeNumber/PrettifyLargeNumber.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC } from 'react';
+import millify from 'millify';
+import { Tooltip } from '@mui/material';
+import { LARGE_NUMBER_PRETTIFIED } from 'utils/testIds';
+
+interface IPrettifyLargeNumberProps {
+    /**
+     * Value to prettify
+     */
+    value: number;
+    /**
+     * Threshold above which the number will be prettified. Values lower than this will just have comma separators added
+     * @default 1_000_000
+     */
+    threshold?: number;
+    /**
+     * The number of significant figures
+     * @default 2
+     */
+    precision?: number;
+}
+
+export const PrettifyLargeNumber: FC<IPrettifyLargeNumberProps> = ({
+    value,
+    threshold = 1_000_000,
+    precision = 2,
+}) => {
+    let prettyValue: string;
+    let showTooltip = false;
+
+    if (value < threshold) {
+        prettyValue = value.toLocaleString();
+    } else {
+        prettyValue = millify(value, { precision });
```

Comment:
Maybe it could be interesting to expose `MillifyOptions` in the props interface, but this is something that can be done later whenever we need it 🤔

Reply:
Yup, that makes sense. Thought of keeping only the required and important ones for this component to begin with